tools/ci/python/program.py

```diff
@@ -0,0 +1,144 @@
+        count, task = tasks_client.list_tasks(
+            (Task.status == TaskStatus.COMPLETED) & (Task.session_id == session_id),
+            page=0,
+            page_size=1,
+            sort_field=Task.processed_at,
+            sort_direction=which,
+        )
+        if return_count:
+            return count, task[0]
+        else:
+            return task[0]
+
+    first_task = get_first_or_last_task(ThroughputGetterHelper.FIRST)
+    tasks_nb, last_task = get_first_or_last_task(
+        ThroughputGetterHelper.LAST, return_count=True
+    )
+
+    logging.info(
+        f"Nb of tasks: {tasks_nb}, First task started at: {first_task.started_at}, Last task ended at: {last_task.ended_at}"
+    )
+
+    return {
+        "tasks_nb": tasks_nb,
+        "throughput": tasks_nb
+        / (last_task.ended_at - first_task.started_at).total_seconds(),
+    }
+
+
+def poll_session_ending(
+    session_id: str, grpc_channel: grpc.Channel, polling_limit: float
+):
+    sessions_client = ArmoniKSessions(grpc_channel)
+    timeout_date = datetime.datetime.now() + datetime.timedelta(seconds=polling_limit)
+    logging.info(
+        f"Starting to wait for session {session_id} to end at : {datetime.datetime.now()}, will end polling at {timeout_date}."
+    )
+
+    while datetime.datetime.now() < timeout_date:
+        session_status = sessions_client.get_session(session_id).status
+        if session_status != SessionStatus.PURGED:
+            logging.info(
+                f"{datetime.datetime.now()} : Waiting for session {session_id} to end"
+            )
+            time.sleep(5)
+        else:
+            logging.info(f"Session {session_id} finished.")
+            return
+    logging.info("Timeout date has been exceeded.")
+
+
+def main(session_name: str, grpc_endpoint: str, polling_limit: float, path: str) -> str:
+    try:
+        with grpc.insecure_channel(f"{grpc_endpoint}:5001") as channel:
+            session_id = get_session_id_by_name(session_name, channel)
+            poll_session_ending(session_id, channel, polling_limit)
+            session_stats = get_session_stats(session_id, channel)
+
+    except:  # noqa: E722
+        logging.info(
+            f"Session {session_name} was not found or gRPC channel located at {grpc_endpoint} cannot be reached."
+        )
+        return
+
+    logging.info(
+        f"Throughput for session named '{session_name}' with id {session_id}: {session_stats['throughput']} Tasks per second"
+    )
+
+    output_path = write_json_output(session_id, session_stats, path)
+    return output_path
+
+
+def write_json_output(session_id: str, session_stats: dict, path: str = "") -> str:
+    json_bench = [
+        {
+            "name": "HTC Mock Throughput",
+            "unit": "Task per second",
+            "value": session_stats["throughput"],
+        }
+    ]
+
+    json_string = json.dumps(json_bench, indent=2)
+
+    logging.info(f"JSON output written in benchmark.json:\n {json_string}")
```

Comment:
what do you need to put in log an information which is already in a file?

Reply:
In order to have a first overview on what is supposed to be written in the json file. Mainly to ensure the conformity of the JSON structure but also to see whether the data to be written is absurd.